app/services/transaction_service/store/transaction.rb

```diff
@@ -217,10 +218,17 @@ def is_booking?(tx_data)
     tx_data[:booking_fields] && tx_data[:booking_fields][:start_on] && tx_data[:booking_fields][:end_on]
   end
 
-  def booking_duration(booking_data)
+  def booking_duration(booking_data, unit_type)
     start_on = booking_data[:start_on]
     end_on = booking_data[:end_on]
-    DateUtils.duration_days(start_on, end_on)
+
+    if unit_type == :day
+      DateUtils.duration_days(start_on, end_on)
+    elsif unit_type == :night
+      DateUtils.duration_nights(start_on, end_on)
+    else
+      raise ArgumentError.new("Unknown unit_type for a booking: #{unit_type}")
+    end
```

Comment:
The calculation here shouldn't happen at all according to our model that we just sketched. And even if it happens unit_type shouldn't matter because start_on and end_on here are always stored the same way (end exclusive).